samples/csharp/MicrosoftEduGraphSamples/Workflows/SubmissionWorkflow.cs

```diff
@@ -33,36 +33,36 @@ public void ReassignWorkflow()
                 var graphClient = MicrosoftGraphSDK.GraphClient.GetDelegateClient(_config["tenantId"], _config["appId"], _config["teacherAccount"], _config["password"]);
 
                 // Teacher creates a new assignment
-                var assignment = MicrosoftGraphSDK.Assignment.Create(graphClient, _config["classId"]);
-                assignmentId = assignment.Result.Id;
-                Console.WriteLine($"Assignment created successfully {assignment.Result.Id} in state {assignment.Result.Status}");
+                var assignment = await MicrosoftGraphSDK.Assignment.CreateAsync(graphClient, _config["classId"]);
+                assignmentId = assignment.Id;
+                Console.WriteLine($"Assignment created successfully {assignment.Id} in state {assignment.Status}");
 
                 // Teacher publishes the assignment to make it appears in the student's list
-                assignment = MicrosoftGraphSDK.Assignment.Publish(graphClient, _config["classId"], assignmentId);
-                Console.WriteLine($"Assignment {assignment.Result.Id} publish in process");
+                assignment = await MicrosoftGraphSDK.Assignment.PublishAsync(graphClient, _config["classId"], assignmentId);
+                Console.WriteLine($"Assignment {assignment.Id} publish in process");
 
                 // Verify assignment state, publish is completed until state equals "Assigned"
-                while (assignment.Result.Status.ToString() != "Assigned" && retries <= MAX_RETRIES)
+                while (assignment.Status.ToString() != "Assigned" && retries <= MAX_RETRIES)
```

Comment:
Just noticing: is this `.ToString()` needed? I would expect to just compare to Status or if it's an enum, use enum for RHS of comparison

Reply:
It is not possible as the data type is different for the SDK, we need to convert it to string.

Reply:
Updated the changes